JASP-Engine/JASP/R/regressionlinearbayesian.R

```diff
@@ -15,587 +15,360 @@
-				InclusionProbabilitiesPlot = plotInclusionProbabilities,
-				plotQQplot = plotQQplot,
-				coefficentsPlots =
-					list(
-						title = "Coefficent plots",
-						collection = plotCoefficientsPosterior
-					)
-			)
-
-	}
-
-	# Set keep and the state
-	if (perform == "init") {
-		statusAnalysis <- "inited"
-		keep <- state$keep
-	} else { #run
-		statusAnalysis <- "complete"
-		keep <- c(
-			plotPosteriorLogOdds$data,
-			sapply(plotCoefficientsPosterior, function(x) x$PosteriorCoefficients$data),
-			plotResidualsVsFitted$data,
-			plotModelProbabilities$data,
-			plotModelComplexity$data,
-			plotInclusionProbabilities$data,
-			postSummaryPlot$data,
-			plotQQplot$data
-		)
-
-		state <- list(
-			options = options,
-			bas_obj = bas_obj,
-			postSummary = postSummary,
-			postSummaryTable = postSummaryTable,
-			postSummaryPlot = postSummaryPlot,
-			descriptives = descriptives,
-			plotPosteriorLogOdds = plotPosteriorLogOdds,
-			plotCoefficientsPosterior = plotCoefficientsPosterior,
-			plotResidualsVsFitted = plotResidualsVsFitted,
-			plotModelProbabilities = plotModelProbabilities,
-			plotModelComplexity = plotModelComplexity,
-			plotInclusionProbabilities = plotInclusionProbabilities,
-			plotQQplot = plotQQplot,
-			status = status,
-			keep = keep
-		)
-	}
-
-	if (!is.null(state))
-		attr(state, "key") <- stateKey
-
-	return(list(
-		keep = keep,
-		results = results,
-		status = statusAnalysis,
-		state = state)
-	)
+RegressionLinearBayesian <- function(jaspResults, dataset = NULL, options, ...) {
+  
+  jaspResults$title <- "Bayesian Linear Regression"
+  
+  status <- .basregSetStatus(options)
+  
+  dataset <- .basregReadData(dataset, options, status)
+  
+  .basregCheckErrors(dataset, options, status)
+  
+  basregResults <- .basregComputeModel(dataset, jaspResults, options, status)
+  status <- basregResults$status
+  basregModel <- basregResults$model
+  
+  .basregTableModelComparison(basregModel, jaspResults, options, status)
+  .basregTableDescriptives(dataset, jaspResults, options, status)
+  
+  posteriorSummary <- .basregComputePosteriorSummary(basregModel, dataset, jaspResults, options, status)
+  
+  .basregTablePosteriorSummary(posteriorSummary, basregModel, jaspResults, options, status)
+  .basregPlotPosteriorSummary(posteriorSummary, jaspResults, options, status)
+  .basregplotsPosteriorDistribution(posteriorSummary, basregModel, jaspResults, options, status)
+  
+  .basregPlotResidualsVsFitted(basregModel, jaspResults, options, status)
+  .basregPlotModelProbabilities(basregModel, jaspResults, options, status)
+  .basregPlotModelComplexity(basregModel, jaspResults, options, status)
+  .basregPlotInclusionProbabilities(basregModel, jaspResults, options, status)
+  .basregPlotQQ(basregModel, jaspResults, options, status)
+  
+  # this function still needs to be rewritten to ggplot (start in .plotImage.basReg):
+  .basregPlotPosteriorLogOdds(basregModel, jaspResults, options, status)
```

Comment:
This means there will be a hard crash (not an R crash) when a user tries to copy this plot as pdf or eps. So for the next release this should be fixed!

Reply:
Mkay. Maybe we can just omit the plot if no one gets it anyway...